src/pfe/portal/modules/Project.js

```diff
@@ -148,9 +166,50 @@ module.exports = class Project {
     return {}
   }
 
+  async getPerfDashPath() {
+    const isPerfDashAvailable = this.injectMetrics || this.isOpenLiberty || (await this.checkIfMetricsAvailable());
+    return isPerfDashAvailable
+      ? `/performance/charts?project=${this.projectID}`
+      : null;
+  }
+  
+  async getMetricsDashHost() {
+    const isMetricsDashAvailable = this.injectMetrics || this.isOpenLiberty || (await this.checkIfMetricsAvailable());
+    if (!isMetricsDashAvailable) {
+      return null;
+    }
+    
+    const shouldShowMetricsDashHostedOnProject = !this.injectMetrics && (await this.checkIfMetricsAvailable());
+    return shouldShowMetricsDashHostedOnProject
+      ? METRICS_DASH_HOST.project
+      : METRICS_DASH_HOST.performanceContainer;
+  }
+  
+  async getMetricsDashPath() {
+    const metricsDashHost = await this.getMetricsDashHost();
+    if (!metricsDashHost) {
+      return null;
+    }
+    
+    if (metricsDashHost === METRICS_DASH_HOST.project) {
+      return `/${pathsToUserHostedMetricsDashboards[this.language]}?theme=dark`;
+    }
+
+    if (metricsDashHost === METRICS_DASH_HOST.performanceContainer) {
+      return `/performance/monitor/dashboard/${this.language}?theme=dark&projectID=${this.projectID}`
+    }
+    
+    throw new Error(`unknown metricsDashHost: ${metricsDashHost}`);
+  }
 
   async checkIfMetricsAvailable() {
     let isMetricsAvailable;
+
+    if (typeof this.metricsAvailable === 'boolean') {
+      // Already checked
+      return this.metricsAvailable;
+    }
+
```

Comment:
I added the above check because now that `retrieveProject` calls this function, this function gets called a lot, meaning we were scanning the project's build file a lot (see below)

Reply:
it sounds like we're caching that status now, which is obviously the right thing to do (so :+1: for that) but will this cached metricsAvailable be cleared if the project's package.json/pom.xml changes ?

Reply:
great spot, I believe this change stops `metricsAvailable` updating in that situation. I'll think of a solution. Maybe `upload` or `uploadEnd` should update `metricsAvailable` if it notices the project's package.json/pom.xml changing

Reply:
I've opened an issue to reinstate our tests for this behaviour https://github.com/eclipse/codewind/issues/2104